### ISSUE TITLE
deep namespace: validate when imported just as a name

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#189]
given:
```js
// ./module/foo.js
export function something() { /* whatever */ }
```
```js
// ./module/index.js
export * as foo from './foo'
```
should report on
```js
// works
import * as mod from './module'
mod.foo.somethingElse() // 'somethingElse' not found in deeply imported namespace 'mod.foo'.

// should work
import { foo } from './module'
foo.somethingElse() // 'somethingElse' not found in deeply imported namespace 'foo'.
```
